### ISSUE TITLE
New rule: check that loop variable is not changed in loop body

### ISSUE THREAD

**Issue by CZechBoY** [eslint/eslint#10993]
Hi,
I would like to have new rule, that checks if for-loop-variable (that is used as condition) is not changed in loop body, or at least is not write-used in nested loop.

**The problem you want to solve.**
```js
for (var someVar = 0; someVar < 10; someVar++) {
    for (someVar = 0; someVar < 20; someVar++) {
        // ... any code ...
    }
}
```

**Your take on the correct solution to problem.**
Should warn about using `someVar` in nested for-loop, which is used in parent loop.

**Comment by nzakas** [comment]
Thanks for the suggestion. This seems like a very specific problem that I'm not sure makes sense as part of ESLint. However, you can certainly make your own rule as a plugin if you feel this is important for your codebase.

**Comment by CZechBoY** [comment]
ok, thanks.
